### PR TITLE
Pre-cache version field changes from ocp-release to sha-256 container image digests 

### PR DESCRIPTION
**Pull request by alosadagrande**
I found problems when using pre-cache feature when the version of the ocp-release to be used was not set as container image digest. The pre-cache tasks were done successfully but when it comes to replacing the image  ( (clusterversion.spec.desiredUpdate.image) of the Clusterversion object with a non digest container image the cluster-operator does not trust the image.

Probably it can be solved by setting the force field of the Clusterversion manifest to true, but.... I do not think it is the correct way.

When setting the container image digest in the CGU CR, both the pre-cache and upgrade tasks were executed successfully.

This PR basically modifies README where it is explained how to easily obtain the digest using oc binary and modifies the examples spread over the repo. Also, it changes a comment in the code.

cc/ @vitus133 

Signed-off-by: Alberto Losada <alosadag@redhat.com>

### CODE REVIEW

**Comment by alosadagrande** [comment]
cc/ @vitus133

File: config/pre-cache/README.md [hunk 1]
```diff
@@ -7,16 +7,33 @@ Supported  overrides:
...
+
+
 Example:
 
 ```yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: cluster-group-upgrade-overrides
+  - du-upgrade-platform-upgrade-candidate-channel
```
Comment (vitus133) [inline config/pre-cache/README.md:33]: Is this line intentional?

File: controllers/precache.go [hunk 2]
```diff
@@ -565,7 +565,7 @@ func (r *ClusterGroupUpgradeReconciler) getPrecacheJobTemplateData(
 
 // getPrecacheSoftwareSpec: Get precaching payload spec for a cluster. It consists of
 //    	several parts that together compose the precaching workload API:
-//		1. platform.image (e.g. "quay.io/openshift-release-dev/ocp-release:<tag>").
+//		  1. platform.image as a SHA-256 container image digest (e.g. "quay.io/openshift-release-dev/ocp-release:<sha256-digest>").
```
Comment (vitus133) [inline controllers/precache.go:568]: Just the alignment
```suggestion
//		1. platform.image as a SHA-256 container image digest (e.g. "quay.io/openshift-release-dev/ocp-release:<sha256-digest>").
```

File: samples/clustergroupupgrade1.yaml [hunk 3]
```diff
@@ -20,6 +20,6 @@ spec:
     channel: stable-4.8
     desiredUpdate:
       version: 4.8.1
-      image: quay.io/openshift-release-dev/ocp-release:4.8.1-x86_64
+      image: quay.io/openshift-release-dev/ocp-release@sha256:d56d85d0f773db3046bad3858a89c2deaf53c14703d58e3e9966dbda1331856b
```
Comment (vitus133) [inline samples/clustergroupupgrade1.yaml:23]: @irinamihai OK with you

File: samples/clustergroupupgrade3.yaml [hunk 4]
```diff
@@ -20,7 +20,7 @@ spec:
     channel: stable-4.8
     desiredUpdate:
       version: 4.8.5
-      image: quay.io/openshift-release-dev/ocp-release:4.8.5-x86_64
+      image: quay.io/openshift-release-dev/ocp-release@sha256:7047acb946649cc1f54d98a1c28dd7b487fe91479aa52c13c971ea014a66c8a8
```
Comment (vitus133) [inline samples/clustergroupupgrade3.yaml:23]: @irinamihai ok with you?

**Review by vitus133 — COMMENTED** [review]
Thanks a lot for this!
Added Irina where it concerns files she owns :)

**Pull request merged**